derp: TestSendFreeze flakes on 32-bit GitHub builder

Our 32-bit builder on GitHub actions sometimes fails with:

```
--- FAIL: TestSendFreeze (0.22s)
Error:     logger.go:33: derp client alice/d2b400d77cdbf575c49c746d3c43c020ec96ab4d56290b30f6910304ca51cd10: adding connection
Error:     logger.go:33: derp client bob/d25257fc97414e0da3fe5c2e5d6b663e3902304cf03b3d42d8d285e4ecaa141d: adding connection
Error:     logger.go:33: derp client cathy/9d3d2ea159684a9b430f3b979226c3164cc75c1530d2f3d47be1040f37b9f736: adding connection
Error:     derp_test.go:402: TEST COMPLETE, cancelling sender
Error:     derp_test.go:404: closing connections
Error:     logger.go:33: derp client alice/d2b400d77cdbf575c49c746d3c43c020ec96ab4d56290b30f6910304ca51cd10: read EOF
Error:     logger.go:33: derp client alice/d2b400d77cdbf575c49c746d3c43c020ec96ab4d56290b30f6910304ca51cd10: removing connection
Error:     logger.go:33: derp client bob/d25257fc97414e0da3fe5c2e5d6b663e3902304cf03b3d42d8d285e4ecaa141d: read EOF
Error:     logger.go:33: derp client bob/d25257fc97414e0da3fe5c2e5d6b663e3902304cf03b3d42d8d285e4ecaa141d: removing connection
Error:     logger.go:33: derp client cathy/9d3d2ea159684a9b430f3b979226c3164cc75c1530d2f3d47be1040f37b9f736: read EOF
Error:     logger.go:33: derp client cathy/9d3d2ea159684a9b430f3b979226c3164cc75c1530d2f3d47be1040f37b9f736: sender failed: use of closed network connection
Error:     logger.go:33: derp client cathy/9d3d2ea159684a9b430f3b979226c3164cc75c1530d2f3d47be1040f37b9f736: removing connection
Error:     derp_test.go:415: alice send to bob: derp.Send: use of closed network connection
FAIL
FAIL	tailscale.com/derp	0.372s
```

--- a/derp/derp_test.go
+++ b/derp/derp_test.go
@@ -409,7 +409,7 @@ func TestSendFreeze(t *testing.T) {
 	for i := 0; i < cap(errCh); i++ {
 		err := <-errCh
 		if err != nil {
-			if errors.Is(err, io.EOF) {
+			if errors.Is(err, io.EOF) || errors.Is(err, net.ErrClosed) {
 				continue
 			}
 			t.Error(err)

